src/main/java/com/commercetools/sync/commons/helpers/BaseSyncStatistics.java

```diff
@@ -0,0 +1,237 @@
+package com.commercetools.sync.commons.helpers;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.netty.util.internal.StringUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import java.util.concurrent.TimeUnit;
+
+import static java.lang.String.format;
+
+public class BaseSyncStatistics {
+    private static final Logger LOGGER = LoggerFactory.getLogger(BaseSyncStatistics.class);
+
+    private String reportMessage;
+    private int updated;
+    private int created;
+    private int failed;
+    private int processed;
+    private long startTime;
+    private long processingTimeInDays;
+    private long processingTimeInHours;
+    private long processingTimeInMinutes;
+    private long processingTimeInSeconds;
+    private long processingTimeInMillis;
+    private String humanReadableProcessingTime;
+
+    /**
+     * Creates a new instance of {@link BaseSyncStatistics} which stores the current time of instantiation in the
+     * {@code startTime} instance variable that will be used later when {@link BaseSyncStatistics#calculateProcessingTime()}
+     * is called to calculate the total time of processing.
+     */
+    public BaseSyncStatistics() {
+        startTime = System.currentTimeMillis();
```

Comment:
In my opinion it's better to provide additional "start timer" method that you could use at the beginning of sync method, rather that initialize `startTime` during object construction. Current initialization of `startTime` can lead to incorrect measurements. It's already misused in `CategorySync` (see my comment in constructor of that class).